harness/harness.go

```diff
@@ -61,13 +61,16 @@ func (h *Harness) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	// Flush any change events and rebuild app if necessary.
-	// Render an error page if the rebuild / restart failed.
-	err := watcher.Notify()
-	if err != nil {
-		atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
-		renderError(w, r, err)
-		return
+	// If app did not start when harness was run then trigger the build to capture the error
+	if h.app==nil {
+		// Flush any change events and rebuild app if necessary.
+		// Render an error page if the rebuild / restart failed.
+		err := h.watcher.Notify()
+		if err != nil {
+			atomic.CompareAndSwapInt32(&lastRequestHadError, 0, 1)
```

Comment:
`CompareAndSwapInt32` is kind of obscure, we should add comments to lines referencing this method to communicate our intent which I believe has to do with syncing up the concurrent change to `lastRequestHadError`